aws-secretsmanager/src/main/java/io/micronaut/aws/secretsmanager/SecretsManagerConfigurationProperties.java

```diff
@@ -56,4 +64,38 @@ public boolean isEnabled() {
     public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
+
+    /**
+     * Sets the secret configuration.
+     *
+     * @param secrets the secret configuration
+     */
+    public void setSecrets(List<Map<String, String>> secrets) {
+        this.secrets = secrets;
+    }
+
+    /**
+     * @return the secret configuration
+     */
+    @Override
+    public List<SecretHolder> getSecrets() {
+        List<SecretHolder> secretHolders = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(secrets)) {
+            for (Map<String, String> secretHolderMap : secrets) {
+                Optional<SecretHolder> secretHolder = convertMapToSecretHolder(secretHolderMap);
+                if (secretHolder.isPresent()) {
+                    secretHolders.add(secretHolder.get());
+                }
+            }
+        }
+        return secretHolders;
+    }
+
+    private Optional<SecretHolder> convertMapToSecretHolder(Map<String, String> secretHolderMap) {
+        if (CollectionUtils.isNotEmpty(secretHolderMap)) {
+            SecretHolder secretHolder = new SecretHolder(secretHolderMap.get("secret"), secretHolderMap.get("prefix"));
+            return Optional.of(secretHolder);
+        }
+        return Optional.empty();
+    }
```

Comment:
You don't need to do this. You can move the type `SecretHolder` (better name `SecretConfiguration`) into a nested class and annotate it with `@EachProperty("secrets")` and then just inject it which would let you delete all of this code

Reply:
@graemerocher could you please clarify what you mean by ".then just inject it"?

Reply:
I think I got it. The problem was that I was defining secrets list member in the outer class with a private modifier versus protected.

Reply:
Done